homeassistant/components/reolink/light.py

```diff
@@ -0,0 +1,153 @@
+"""This component provides support for Reolink light entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from reolink_aio.api import Host
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ColorMode,
+    LightEntity,
+    LightEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkLightEntityDescriptionMixin:
+    """Mixin values for Reolink light entities."""
+
+    is_on: Callable[[Host, int | None], bool]
+    turn_on_off: Callable[[Host, int | None, bool], Any]
+
+
+@dataclass
+class ReolinkLightEntityDescription(
+    LightEntityDescription, ReolinkLightEntityDescriptionMixin
+):
+    """A class that describes light entities."""
+
+    supported: Callable[[Host, int | None], bool] = lambda api, ch: True
+    get_brightness: Callable[[Host, int | None], int] | None = None
+    set_brightness: Callable[[Host, int | None, float], Any] | None = None
```

Comment:
When specifying callable in entity description the norm is to suffix it with `_fn`

Reply:
done